### DISCUSSION TITLE
About python after deployment

### DISCUSSION THREAD

**Discussion by esmo-ts** [deta/cloud-docs#681] [Help]
The error happened locally and I solved by changing "if dtype == np.object" to "if dtype == np.object_" in the local python directory, but when I deployed it the error happened again, the error is not from Deta, it's from Numpy, they're changing that "object" to "object_", so to help me with this it needs to be altered by Deta admins, this is the error log so you can find its path :

Traceback (most recent call last):
  File "/opt/python/detalib/debugger.py", line 142, in wrap
    result = func(event, context)
  File "/var/task/_entry.py", line 12, in handler
    import main  # noqa
  File "/var/task/main.py", line 9, in <module>
    app = FastAPI_CSV("./nhl_elo.csv")
  File "/opt/python/fastapi_csv/applications.py", line 118, in __init__
    type_ = dtype_to_type(dtype)
  File "/opt/python/fastapi_csv/applications.py", line 35, in dtype_to_type
    if dtype == np.object:
  File "/opt/python/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'object'

**Comment by etrotta** [comment]
Numpy has deprecated `numpy.object` in favor of `numpy.object_` as of [version 1.20](https://numpy.org/doc/stable/release/1.20.0-notes.html), and fully removed support for `numpy.object` as of [version 1.24](https://numpy.org/doc/stable/release/1.24.0-notes.html).
That said, it is up to you to specify which version you are using (on your requirements.txt), not up to deta.
Remember to use `pip freeze` to pin versions instead when deploying stuff.

**Reply by esmo-ts** [reply]
It's not up to me, I installed the latest version of Numpy on my computer and solved it that way, I did the same with the requirements and it still raise the same error on Deta

**Comment by begoon** [comment]
You can only write to `/tmp`. Everything else is read-only.
